framework/PluginsBaseClasses/PostProcessorPluginBase.py

```diff
@@ -0,0 +1,129 @@
+      @ In, **kwargs, dict, is a dictionary that contains the information coming from the sampler,
+          a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}.
+          Not used for PostProcessor, and {'SampledVars':{'prefix':'None'}, 'additionalEdits':{}}
+          is used during "Step" "PostProcess" handling
+      @ Out, inputDs, list, list of data set that will be directly used by the "PostProcessor.run" method.
+    """
+    #### TODO: This method probably need to move to PostProcessor Base Class when we have converted
+    #### all internal PostProcessors to use Dataset
+
+    ## Type 1: DataObjects => Dataset
+    ## Type 2: File => File
+    ## Type 3: HDF5 => ?
+    assert type(inputObjs) == list
+    inputDs = []
+    for inp in inputObjs:
+      if isinstance(inp, Files.File):
+        inputDs.append(inp)
+      elif isinstance(inp, DataObject.DataObject):
+        # Current accept two types: 1) 'dict', 2) 'xrDataset'
+        # Set default to 'dict', this is consistent with current post-processors
+        outType = kwargs.get('outType', 'dict')
+        inputDs.append(inp.asDataset(outType=outType))
+      elif isinstance(inp, Database):
+        self.raiseAnError(IOError, "Database", inp.name, "can not be handled directly by this Post Processor")
+      else:
+        self.raiseAnError(IOError, "Unknown input is found", str(inp))
+    return inputDs
+
+
+  ##################################################
+  # Plugin APIs
+  ##################################################
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super().getInputSpecification()
+    return inputSpecification
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      This function is used to initialize the plugin, i.e. set up working dir,
+      call the initializePlugin method from the plugin
+      @ In, runInfo, dict, it is the run info from the jobHandler
+      @ In, inputs, list, it is a list containing whatever is passed with an input role in the step
+      @ In, initDict, dict, optional, dictionary of all objects available in the step is using this model
+    """
+    super().initialize(runInfo, inputs, initDict)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the common parts of the model parameter input.
+      @ In, paramInput, InputData.ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
```

Comment:
My only thought from the plugin side is the somewhat confusing usage of the term "input".

In `handleInput` it means "handle input XML from the RAVEN input file" whereas in `createNewInput` it means "prepare the input for a single postprocessor run". I wonder how (or even if) we could approach that to make it more clear for plugin developers.